strapi/strapi
# [DTS] Allow schema check bypass; remove data that cannot be transfered

### What does it do?

The schema check at the beginning of a transfer is no longer a hard stop. Instead, it accepts a callback for confirmation, so that the command actions (import, export, transfer) can prompt the user to confirm that they may lose data in the case of a differing schema (or continue automatically with --force)

Functionality should be complete and can be reviewed, but we still need to document this behavior.

### Why is it needed?

To allow for transfers where there may be data loss (such as a Community Edition Strapi without review workflows pushing data to an Enterprise Edition Strapi) to happen anyway

### How to test it?

Do a transfer from a CE Strapi to an EE Strapi
  - You should be warned that there is a schema difference 
  - You should be prompted to continue without transfering that data
    - if you confirm, the transfer should succeed without transfering that data
    - if you refuse, the transfer should halt

Do a transfer from a EE strapi to a CE Strapi
  - You should be warned that there is a schema difference 
  - You should be prompted to continue without transfering that data
    - if you confirm, the transfer should succeed without transfering that data
    - if you refuse, the transfer should halt
  
Do an export from an instance of Strapi that contains several content types, including a relation to a content type that will be deleted
- Delete and/or add fields from content type schema
- Delete and/or add an entire content type schema
- Do a transfer from one Strapi to the other (and vice versa)
  - You should be warned that there is a schema difference 
  - You should be prompted to continue without transfering that data
    - if you confirm, the transfer should succeed without transfering that data
    - if you refuse, the transfer should halt

### Related issue(s)/PR(s)

Fixes https://github.com/strapi/strapi/issues/16573
